src/ros_filter.cpp

```diff
@@ -1898,6 +1901,9 @@ namespace RobotLocalization
   template<typename T>
   void RosFilter<T>::aggregateDiagnostics(diagnostic_updater::DiagnosticStatusWrapper &wrapper)
   {
+    // This mutex prevents aggregateDiagnostics and addDiagnostic from clashing
+    boost::mutex::scoped_lock lock(diagnosticMapsMutex_);
```

Comment:
I agree, I could have added a new scope around the for loops and restricted the mutex to that location but I figured that starts to make larger deltas.